File: QE-selenium python/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/dynamic-controls")
print(driver.title)

checkbox = driver.find_element(By.XPATH, "//input[@type='checkbox']")
toggle_button = driver.find_element(By.ID, "toggleCheckbox")

toggle_button.click()
wait = WebDriverWait(driver, 10)
wait.until(EC.invisibility_of_element(checkbox))

toggle_button.click()
checkbox = wait.until(EC.visibility_of_element_located((By.XPATH, "//input[@type='checkbox']")))
checkbox.click()

driver.quit()

File: QE selenium java/Activity11.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;
public class Activity11 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-controls");
        System.out.println(driver.getTitle());

        WebElement checkbox = driver.findElement(By.xpath("//input[@type='checkbox']"));
        WebElement toggleButton = driver.findElement(By.id("toggleCheckbox"));

        toggleButton.click();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.invisibilityOf(checkbox));

        toggleButton.click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@type='checkbox']")));
        driver.findElement(By.xpath("//input[@type='checkbox']")).click();

        driver.quit();
    }
}

